generator/internal/language/client.go

```diff
@@ -33,17 +33,49 @@ func (p *mustacheProvider) Get(name string) (string, error) {
 	return p.impl(filepath.Join(p.dirname, name) + ".mustache")
 }
 
-func GenerateClient(model *api.API, codec Codec, outdir string) error {
-	data := newTemplateData(model, codec)
-	var context []any
-	context = append(context, data)
-	if languageContext := codec.AdditionalContext(model); languageContext != nil {
-		context = append(context, languageContext)
+func GenerateClient(model *api.API, language, outdir string, options map[string]string) error {
+	var (
+		context        []any
+		provider       TemplateProvider
+		generatedFiles []GeneratedFile
+	)
+	switch language {
+	case "rust":
+		codec, err := NewRustCodec(outdir, options)
+		if err != nil {
+			return err
+		}
+		if err := codec.Validate(model); err != nil {
+			return err
+		}
+		data := newRustTemplateData(model, codec)
+		context = append(context, data)
+		if languageContext := codec.AdditionalContext(model); languageContext != nil {
+			context = append(context, languageContext)
+		}
+		provider = codec.TemplatesProvider()
+		generatedFiles = codec.GeneratedFiles()
```

Comment:
Maybe this is already part of the cleanup PRs, but it feels that this should become:

```
     case "rust":
        return GenerateRust(outdir, options, model)
```

maybe with the code starting at line 78 refactored to a helper function?

Reply:
yup, will do in a follow up PR